### PR TITLE
[Improvement] Add object fit context and Polyfill for IE

### PR DESCRIPTION
**Pull request by beau-charman**
Dependant of https://github.com/aesop/aesop-web-ui/pull/3931

### CODE REVIEW

**Comment by MkMan** [comment]
Seeing that we're dropping support for IE, is this still needed?

**Comment by beau-charman** [comment]
Short answer yes, but still finding that out

File: src/contexts/ObjectFitContext/ObjectFitContext.types.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import type { ComponentWithChildren } from '~/types';
```
Comment (MkMan) [inline src/contexts/ObjectFitContext/ObjectFitContext.types.ts:1]: Should we go back to using `FC` and `VFC`? The Storybook documentation was really helpful when figuring out how to use the components.

File: src/components/Image/Image.hooks.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { useEffect } from 'react';
+
+const useObjectFitPolyfill = (
+  image: any,
```
Comment (MkMan) [inline src/components/Image/Image.hooks.ts:4]: Is `image` a ref?

File: src/contexts/ObjectFitContext/ObjectFitContext.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+const ObjectFitContextProvider: ObjectFitContextProviderType = ({
+  children,
+}) => (
+  <ObjectFitContext.Provider value={true}>{children}</ObjectFitContext.Provider>
+);
+
+const useObjectFitContext = (hasObjectFit: boolean): boolean => {
+  const objectFitContext = useContext(ObjectFitContext);
+  return hasObjectFit !== undefined ? hasObjectFit : objectFitContext;
```
Comment (MkMan) [inline src/contexts/ObjectFitContext/ObjectFitContext.tsx:14]: Is this the same as `hasObjectFit ?? objectFitContext`?

File: src/contexts/ObjectFitContext/ObjectFitContext.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React, { createContext, useContext } from 'react';
+import type { ObjectFitContextProviderType } from './ObjectFitContext.types';
+
+const ObjectFitContext = createContext(undefined);
+
+const ObjectFitContextProvider: ObjectFitContextProviderType = ({
```
Comment (MkMan) [inline src/contexts/ObjectFitContext/ObjectFitContext.tsx:6]: I didn't completely understand how/when to use this context provider. Can you please a sample flow/example?

**Pull request closed without merging**